### ISSUE TITLE
Find good hash functions

### ISSUE THREAD

**Issue by texone** [TheFuseLab/VL.Fuse#24]
So far we are using hash without sine which does not work to bad but seems to generate weird patterns on particle stuff
https://www.shadertoy.com/view/4djSRW

This one lists are wide variety of functions 
https://www.shadertoy.com/view/XlGcRh

Lets discuss this

**Comment by everyoneishappy** [comment]
That's a super cool paper!  I'd like to implement more, but probably not until this is solved: https://github.com/TheFuseLab/VL.Fuse/issues/4

**Comment by everyoneishappy** [comment]
In the meantime, what we use now is basically a very fast, but as you found low quality (in terms of no patterns) hash function.  This is 100% fine for most noises built on top, but for the time being for simple random number generation should use the random function provided in FuseCommonNoise.  It just iterates the hash dfunction several times and should still be quite performant.  I'll make a node for it again since we deleted the basis nodes in favor of the swiss army knife 'noise' node.  Can still use that for random too, but in a lot of cases probably don't need things like bias etc so will put a simple one back in

**Comment by everyoneishappy** [comment]
Ok if you pull could you try swapping out the Random[Math] for Random[Noise] and see if it's random enough for you?  If not we could look at using a better hash for that.  Would be good if possible to keep using a cheap one for the procedural noises since you don't get much benefit there from a more expensive function, and it can be called many times in some cases

**Comment by texone** [comment]
For compute I switched to pcg hash function that was suggested in the paper got quality and decent performance, I also like the explanations regarding combination of noise values for better quality so I am wandering if we should provide different hash implementations like noise and people can select?

**Comment by everyoneishappy** [comment]
Yeah the pcg ones looks like the best for random number generation hey?  My impression is if you're just running it once or a couple times per thread there's not really a significant difference performance wise at all.  I guess it's more a difference for example if you're calling it repeatedly in a procedural noise function and then repeating that to raymarch then a cheaper hash might be better used.  In the case of noise (simplex, perlin etc) you don't really need the extra quality in the hash.
